tests/benchmarks/test_mod.py

```diff
@@ -0,0 +1,128 @@
+"""Benchmarks for MOD instruction."""
+
+from ethereum_test_tools import (
+    Alloc,
+    Environment,
+    StateTestFiller,
+    Transaction,
+)
+from ethereum_test_tools import Opcodes as Op
+from ethereum_test_vm import Bytecode, Opcode
+
+
+def pick_best(xs, d):
+    """Find arguments to MOD giving the biggest result."""
+    mods = [x % d for x in xs]
+    index_max = max(range(len(mods)), key=mods.__getitem__)
+    return index_max
+
+
+def evaluate(xs, d):
+    """Evaluate mod chain."""
+    indexes = []
+    for _ in range(750):
+        i = pick_best(xs, d)
+        d = xs[i] % d
+        indexes.append(i)
+    return indexes
+
+
+def test_mod(state_test: StateTestFiller, pre: Alloc):
+    """Benchmark for MOD instruction."""
+    env = Environment()
+    sender = pre.fund_eoa()
+    post = {}
+
+    xs = [
+        43236444494015607440141505080495154399932445654715976568919931169187528239188,
+        105777965273184455192929147742637094743111160210117740086738876889283177791980,
+        65953689421958480284379163031189022104793097678164298920000328544846124204137,
+        50555838962958859666768244375999039852090935345987449008145098846820070840849,
+        113453475245602463688947768647175650503593153393058797795502044670530221350790,
+        108515932427494591044498166562596753417879210560589683295016433069972281228649,
+        107722439603782468610619723710494877133859581454624515618825198520300105650547,
+        106459072643751286647557215675084286830523999822283951411423751004203967971365,
+        114539964541068397851108711591272381832888394470830758825121311279332893192841,
+        57139838140016571538944663171893621132175712073852765048630205338112940186675,
+        88761917728074884944552062209595491369240700019109983916079607448076998287508,
+        104533446445848322405203834135751967481158619549985970925941663895363945051353,
+        101633083962374564205750776617705398522073020934676506995472923219075790864659,
+        101208882220548070829583557488550098768856705701019930941525598172176842112649,
+        62742163047811414741822493362602776277205609331255254122243083067565272681517,
+    ]
+    assert len(xs) == 15
+    d = 1773207472995338907891577153413104418251941371322332249409
+
+    code = Bytecode()
+
+    for x in xs:
+        code += Op.PUSH32(x)
+
+    indexes = evaluate(xs, d)
+
+    for _ in range(100):
+        code += Op.PUSH32(d)
+        for i in indexes:
+            dup = Opcode(0x80 + (15 - i))
+            code += dup + Op.MOD
+        code += Op.POP
+
+    account = pre.deploy_contract(code)
+
+    tx = Transaction(
+        to=account,
+        gas_limit=30_000_000,
+        gas_price=10,
+        sender=sender,
```

Comment:
```suggestion
        sender=sender,
        protected=False,
```
Just so the transaction is successful in Frontier too.